src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java

```diff
@@ -132,23 +137,94 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
 				AELog.debug( e );
 			}
 		}
+
+		if( btn instanceof GuiImgButton )
+		{
+			final boolean backwards = Mouse.isButtonDown( 1 );
+
+			final GuiImgButton iBtn = (GuiImgButton) btn;
+			if( iBtn.getSetting() != Settings.ACTIONS )
+			{
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
+				}
+				else
+				{
+					try
+					{
+						NetworkHandler.instance().sendToServer( new PacketValueConfig( iBtn.getSetting().name(), next.name() ) );
+					}
+					catch( final IOException e )
+					{
+						AELog.debug( e );
+					}
+				}
+
+				iBtn.set( next );
+
+				if( next.getClass() == TerminalStyle.class )
+				{
+					this.reinitalize();
+				}
+			}
+		}
+	}
+
+	private void reinitalize()
+	{
+		this.buttonList.clear();
+		this.initGui();
 	}
 
 	@Override
 	public void initGui()
 	{
+		final int staticSpace = ITEMSTACK_TOP_OFFSET + CANCEL_TOP_OFFSET - 1;
+
+		calculateRows(staticSpace);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) '(' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) ')' is not preceded with whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)